db/db.js

```diff
@@ -294,6 +321,65 @@ async function updateSessionStillCounter(stillcounter, sessionid, locationid) {
   }
 }
 
+async function isOverdoseSuspectedInnosent(session, location, clientParam) {
+  try {
+    const now = new Date(await getCurrentTime())
+    const start_time_sesh = new Date(session.start_time)
+
+    // Current Session duration so far:
+    const sessionDuration = (now - start_time_sesh) / 1000
+
+    // threshold values for the various overdose conditions
+    const still_threshold = location.still_threshold
+    const sessionDuration_threshold = location.duration_threshold
+
+    // number in front represents the weighting
+    const condition2 = 1 * (session.still_counter > still_threshold) // seconds
+    const condition3 = 1 * (sessionDuration > sessionDuration_threshold)
+
+    let alertReason = ALERT_REASON.NOTSET
+
+    if (condition3 === 1) {
+      alertReason = ALERT_REASON.DURATION
+    }
+
+    if (condition2 === 1) {
+      alertReason = ALERT_REASON.STILLNESS
+    }
+
+    const conditionThreshold = 1
+
+    // This method just looks for a majority of conditions to be met
+    // This method can apply different weights for different criteria
+    if (condition2 + condition3 >= conditionThreshold) {
+      // update the table entry so od_flag is 1
+      try {
+        let client = clientParam
+        const transactionMode = typeof client !== 'undefined'
+        if (!transactionMode) {
+          client = await pool.connect()
+        }
+        await client.query('UPDATE sessions SET od_flag = $1, alert_reason = $2  WHERE sessionid = $3', [
+          OD_FLAG_STATE.OVERDOSE,
+          alertReason,
+          session.sessionid,
+        ])
+        if (!transactionMode) {
+          client.release()
+        }
+
+        return true
+      } catch (err) {
+        helpers.log(`Error running od_flag update query`)
+      }
+    }
+
+    return false
+  } catch (e) {
+    helpers.log(`Error running isOverdoseSuspected: ${e}`)
```

Comment:
This should be the `isOverdoseSuspectedInnosent` to match the name of the function.